### DISCUSSION TITLE
Strange crash - anyRPC socket closed in special case

### DISCUSSION THREAD

**Discussion by iknerfS** [faaxm/spix#83] [General]
Hello,

This is for now more a question as and issue.

We are using spix v0.3 and I also tried v0.5 and am getting the same problem.

We integrated spxi into the qt app as described in the readme and are using python for testing ([XMLRPC client built into python]).

Our system:
--------------
- windows 10
- Qt v5.15.2
- using visual studio 2022 + qt creator to use MSVC2019 build toolchain.
- using python v3.10

How we are testing:
----------------------
- Our Qt app integrates the spix and also open additional TCP socket for specific user commands/events.
- we use spix commands/function to execute the python tests.

Our UI application has backend written in C++ and forehand in QML and we are using approaches like:

**header file:**

     `Q_PROPERTY( int testStep READ testStep NOTIFY testStepChanged )`

**qml file:**
```
    ...
    property int guideStep:                         TestSetupCtrl.testStep

    ...

    onGuideStepChanged:                             handleGuideControll()

    ...

    function handleGuideControll()
    {
//        console.log("simple logging ")

        switch (guideStep)
        {
            case TestSetupCtrl.QC_SWAP_GUIDE_STEP_PRECAUTIONS:
                guideControll.guideHeaderLabelText  = qsTr("PRECAUTIONS")
                guideControll.guideLabelText        = qsTr("Wear gloves when running a QC test")
                break
    ...

```

We then added python test cases using spix commands wait_screen, mouseclick, etc. And if we execute mouseclick() which simulates ui button click that changes testStep property then with the above code our test case fails, UI crash. But if I uncomment the above line "console.log" then test executes ok. Is there some timing problem here or did I miss something else? And did you encounter this issue on your side? Let me know if you need more information.

Thanks for the answer, Frenk

**Comment by faaxm** [comment]
Moved it to discussions for now, until we have a better idea of what exactly is going wrong. Unfortunately I haven't encountered this issue myself, but maybe somebody else has?!
